casper/src/main/scala/coop/rchain/casper/engine/BlockReceiverState.scala

```diff
@@ -2,17 +2,18 @@ package coop.rchain.casper.engine
 
 import cats.effect.Sync
 import cats.effect.concurrent.Ref
+import cats.syntax.all._
 
 sealed trait RecvStatus
 case object RecvStarted  extends RecvStatus
 case object RecvFinished extends RecvStatus
 
-final case class BlockReceiverState[MId](state: Map[MId, RecvStatus]) {
-  def contains(m: MId): Boolean                                = state.contains(m)
-  def set(m: MId, status: RecvStatus): BlockReceiverState[MId] = copy(state + (m -> status))
+final case class BlockReceiverState[F[_]: Sync, MId](ref: Ref[F, Map[MId, RecvStatus]]) {
+  def contains(m: MId): F[Boolean]             = ref.get.map(_.contains(m))
+  def set(m: MId, status: RecvStatus): F[Unit] = ref.update(_ + (m -> status))
```

Comment:
We can see this class as `BlockReceiver` component which has the state `Map[MId, RecvStatus]`.

Reply:
@tgrospic, can you explain please what do you mean? I just moved this class into BlockReceiver object and created singleton object for class. Is [it](https://github.com/rchain/rchain/pull/3650/commits/038d93195ca84d79862d8cbcf6b8f0d6c80bcbf8) correct?